eclipse-collections/src/main/java/org/eclipse/collections/impl/stack/immutable/ImmutableNotEmptyStack.java

```diff
@@ -429,7 +430,12 @@ public void each(Procedure<? super T> procedure)
     @Override
     public ImmutableStack<T> takeWhile(Predicate<? super T> predicate)
     {
-        throw new UnsupportedOperationException(this.getClass().getSimpleName() + ".takeWhile() not implemented yet");
+        return Stacks.immutable.withAll(this.asReversed().takeWhile(predicate));
+    }
+
+    public ReverseIterable<T> asReversed()
+    {
+        return ReverseIterable.adapt(this.toList());
     }
```

Comment:
We can't add asReversed here. It's not part of the public API (no @Override) here. as* methods are supposed to be O(1) and this implementation is O(2). Anyway, it's not necessary. We can use asLazy() instead.